tensorrt_llm/_torch/pyexecutor/py_executor_creator.py

```diff
@@ -275,6 +275,12 @@ def create_py_executor(
             # The draft model won't have any draft tokens attached to
             # generation requests when we invoke it autoregressively
             draft_spec_config.max_draft_len = 0
+            is_deepseek_model = model_engine.model.model_config.pretrained_config.architectures[
+                0] == "DeepseekV3ForCausalLM"
+            if is_deepseek_model:
+                draft_model_path = checkpoint_dir
+            else:
+                draft_model_path = spec_config.speculative_model_dir
```

Comment:
Can you instead add some logic in `llm_args.py` so that the `spec_config.draft_model_path` gets defaulted to the main checkpoint path when MTP is being used?

I don't think this is strictly correct right now - other spec algorithms with deepseek can specify different draft model paths